src/Compilers/Core/MSBuildTask/Utilities.cs

```diff
@@ -200,11 +200,16 @@ public static string GenerateFullPathToTool(string toolName)
                     if(assemblyPath != null)
                     {
                         var assemblyDirectory = Path.GetDirectoryName(assemblyPath);
-                        var toolLocalLocation = Path.Combine(assemblyDirectory, toolName);
+                        var desktopToolLocalLocation = Path.Combine(assemblyDirectory, toolName);
+                        var cliToolLocalLocation = Path.Combine(assemblyDirectory, "bincore", toolName);
```

Comment:
I have no idea. Seems to be implemented, and has been for a while? https://github.com/dotnet/coreclr/blob/94d374d22c1bb2b4815a05b5049f39d9b78de9cd/src/mscorlib/src/System/Reflection/Assembly.CoreCLR.cs#L264